eureka-client/src/main/java/com/netflix/discovery/shared/resolver/LegacyClusterResolver.java

```diff
@@ -0,0 +1,115 @@
+ * change.
+ * <h3>Failures</h3>
+ * If there is configuration change (from DNS to server list or reverse), and the new resolver cannot be instantiated,
+ * it will be retried with exponential back-off (see {@link ReloadingClusterResolver}).
+ *
+ * @author Tomasz Bak
+ */
+public class LegacyClusterResolver implements ClusterResolver {
+
+    private static final Logger logger = LoggerFactory.getLogger(LegacyClusterResolver.class);
+
+    private final ClusterResolver delegate;
+
+    public LegacyClusterResolver(EurekaClientConfig clientConfig, String myZone) {
+        this.delegate = new ReloadingClusterResolver(
+                new LegacyClusterResolverFactory(clientConfig, myZone),
+                clientConfig.getEurekaServiceUrlPollIntervalSeconds() * 1000
+        );
+    }
+
+    @Override
+    public String getRegion() {
+        return delegate.getRegion();
+    }
+
+    @Override
+    public List<EurekaEndpoint> getClusterEndpoints() {
+        return delegate.getClusterEndpoints();
+    }
+
+    static class LegacyClusterResolverFactory implements ClusterResolverFactory {
+
+        private final EurekaClientConfig clientConfig;
+        private final String myRegion;
+        private final String myZone;
+
+        LegacyClusterResolverFactory(EurekaClientConfig clientConfig, String myZone) {
+            this.clientConfig = clientConfig;
+            this.myRegion = clientConfig.getRegion();
+            this.myZone = myZone;
+        }
+
+        @Override
+        public ClusterResolver createClusterResolver() {
+            ClusterResolver newResolver;
+            if (clientConfig.shouldUseDnsForFetchingServiceUrls()) {
+                newResolver = new DnsTxtRecordClusterResolver(
+                        myRegion,
+                        clientConfig.getEurekaServerDNSName(),
+                        true,
+                        Integer.parseInt(clientConfig.getEurekaServerPort()),
+                        false,
+                        clientConfig.getEurekaServerURLContext()
+                );
+                newResolver = new ZoneAffinityClusterResolver(newResolver, myZone, clientConfig.shouldPreferSameZoneEureka());
+            } else {
+                // FIXME Not randomized in the EndpointUtils.getServiceUrlsFromConfig, and no zone info to do this here
+                newResolver = new StaticClusterResolver(myRegion, createEurekaEndpointsFromConfig());
+            }
+
+            return newResolver;
+        }
+
+        private List<EurekaEndpoint> createEurekaEndpointsFromConfig() {
+            List<String> serviceUrls = EndpointUtils.getServiceUrlsFromConfig(clientConfig, myZone, clientConfig.shouldPreferSameZoneEureka());
+            List<EurekaEndpoint> endpoints = new ArrayList<>(serviceUrls.size());
+            for (String serviceUrl : serviceUrls) {
+                try {
+                    URI serviceURI = new URI(serviceUrl);
```

Comment:
I am not sure if I understand. Host part will be identical to what was provided in service URI, without further resolve. We could provide ClusterResolver decorator that would resolve any host name to its IP value, to avoid transient DNS issues.